dd4t-api/src/main/java/org/dd4t/core/util/TCMURI.java

```diff
@@ -49,20 +55,26 @@ public TCMURI(int publicationId, int itemId, int itemType, int version) {
         this.version = version;
     }
 
+    public TCMURI(Namespace namespace, int publicationId, int itemId, int itemType, int version) {
+        this(publicationId, itemId, itemType, version);
+        this.namespace = namespace;
+    }
+
     public static boolean isValid(String tcmUri) {
-        return tcmUri != null && tcmUri.startsWith(URI_NAMESPACE);
+        return tcmUri != null && startsWithNamespace(tcmUri);
     }
 
     protected void load(String uriString) throws ParseException {
         if (uriString != null) {
-            int namespaceLength = URI_NAMESPACE.length();
-            int uriStringLength = uriString.length();
-            int currentPosition = uriStringLength;
-            if ((uriStringLength < namespaceLength) || (!uriString.startsWith(URI_NAMESPACE))) {
-                throw new ParseException("URI string " + uriString + " does not start with " + URI_NAMESPACE, currentPosition);
+            int colonCharAt = uriString.indexOf(COLON);
+            int currentPosition = uriString.length();
+            if ((colonCharAt < 0) || !startsWithNamespace(uriString)) {
+                throw new ParseException("URI string " + uriString + " does not start with '" + TCM.getValue() + "' or '"
+                        + ISH.getValue() + "'", currentPosition);
             }
 
-            String uri = uriString.substring(namespaceLength, uriStringLength);
+            this.namespace = Namespace.valueOf(uriString.substring(0, colonCharAt).toUpperCase());
```

Comment:
Replace the validating code and the namespace resolving code with a static map in enum. http://codereview.stackexchange.com/questions/46175/java-enum-containing-a-hash-map-for-look-up